aspects/schema.go

```diff
@@ -794,3 +845,19 @@ func (v *arraySchema) parseConstraints(constraints map[string]json.RawMessage) e
 }
 
 func (v *arraySchema) expectsConstraints() bool { return true }
+
+type validationError struct {
```

Comment:
we might want to export this because at some point we might want to translate pieces of path back from storage paths to request paths?